spec/rubocop/cop/rspec/described_class_spec.rb

```diff
@@ -338,6 +338,14 @@ module SomeGem
         RUBY
       end
     end
+
+    it 'ignores a class with a dynamic namespace' do
+      expect_no_offenses(<<~RUBY)
+        describe MyClass do
+          subject { (("MyClass" + "3").constantize)::MyClass }
```

Comment:
This example still seems a bit far fetched 😅

Reply:
I’ll see in real-world-rspec if there’s any code that hits the branch you’re looking at.

Reply:
Not just far fetched, a little hard to find! A good puzzle. 

I buried this in the updated description above, but the way we're getting past the case statement here is just by having a `:begin` node from the parentheses. 

I replaced the example with a simpler parenthesized method call. 

The weird thing is: 
` subject { method_that_returns_a_class::MyClass }` 
does get handled. 

I'm not certain how `method_that_returns_a_class` or `(method_that_returns_a_class)` ought to be corrected, but maybe they should be handled identically.

Reply:
`subject { method_that_returns_a_class::MyClass }` makes `namespace` a send type, whereas `subject { (method_that_returns_a_class)::MyClass }` makes `namespace` a begin type. So maybe the test case we want is just the existing `foo::CONST` one, but with parentheses around `foo`?